packages/gatsby/src/redux/actions/internal.ts

```diff
@@ -350,3 +357,58 @@ export const deleteNodeManifests = (): IDeleteNodeManifests => {
     type: `DELETE_NODE_MANIFESTS`,
   }
 }
+
+export const createJobV2FromInternalJob = (
+  internalJob: InternalJob
+): ICreateJobV2FromInternalAction => (
+  dispatch,
+  getState
+): Promise<Record<string, unknown>> => {
+  const jobContentDigest = internalJob.contentDigest
+  const currentState = getState()
+
+  // Check if we already ran this job before, if yes we return the result
+  // We have an inflight (in progress) queue inside the jobs manager to make sure
+  // we don't waste resources twice during the process
+  if (
+    currentState.jobsV2 &&
+    currentState.jobsV2.complete.has(jobContentDigest)
+  ) {
+    return Promise.resolve(
+      // @ts-ignore we check if `complete` map has result
+      currentState.jobsV2.complete.get(jobContentDigest)!.result
+    )
+  }
+
+  const inProgressJobPromise = getInProcessJobPromise(jobContentDigest)
+  if (inProgressJobPromise) {
+    return inProgressJobPromise
+  }
+
+  dispatch({
+    type: `CREATE_JOB_V2`,
+    payload: {
+      job: internalJob,
+    },
+    plugin: { name: internalJob.plugin.name },
```

Comment:
This is only to ensure https://github.com/gatsbyjs/gatsby/blob/06dfa9c6d8c43299298383404e222294fc8cc6ad/packages/gatsby-plugin-sharp/src/gatsby-node.js#L127-L128 keeps working